File: Go/350_intersection_of_two_arrays_ii.go
func intersect(nums1 []int, nums2 []int) []int {
  sort.Ints(nums1)  
  sort.Ints(nums2)
  res := []int{}
  counter1, counter2 := 0, 0
  for {
    if nums1[counter1] < nums2[counter2] {
      counter1++
    } else if nums1[counter1] > nums2[counter2] {
      counter2++
    } else {
      res = append(res, nums1[counter1])
      counter1++
      counter2++
    }

    if counter1 == len(nums1) || counter2 == len(nums2) {
      break
    }
  }

  return res
}

File: Elixir/350_intersection_of_two_arrays_ii.ex
defmodule Solution do
  @spec intersect(nums1 :: [integer], nums2 :: [integer], res :: [integer]) :: [integer]
  defp intersect(_, [], res), do: res
  defp intersect([], _, res), do: res
  defp intersect([a | b], [a | d], res), do: intersect(b, d, res ++ [a])
  defp intersect([a | b], [c | d], res) when a < c, do: intersect(b, [c | d], res)
  defp intersect([a | b], [c | d], res) when a > c, do: intersect([a | b], d, res)

  @spec intersect(nums1 :: [integer], nums2 :: [integer]) :: [integer]
  def intersect(nums1, nums2) do
    nums1 = nums1 |> Enum.sort(:asc)
    nums2 = nums2 |> Enum.sort(:asc)
    intersect(nums1, nums2, [])
  end
end
